portfolio/src/main/webapp/projects/commentart/scripts/comment.js

```diff
@@ -0,0 +1,68 @@
+class Comment {
+  /**
+   * @param {string} commenterName 
+   * @param {string} commentMsg Commenter's message.
+   * @param {number} sentiment Either 1 (positive) or -1 (negative).
+   * @param {RepliesManager} repliesManager Manager that manages all associated replies.
+   */
+  constructor(commenterName, commentMsg, sentiment, repliesManager) {
+    this.commenterName = commenterName;
+    this.commentMsg = commentMsg;
+    this.sentiment = sentiment;
+    this.repliesManager = repliesManager;
+  }
+
+  /** Converts a comment that is in .json format to a `Comment` object. */
+  static toComment(commentJson) {
+    const commenterName = commentJson.name;
+    const commentMsg = commentJson.commentMsg;
+    const commentSentiments = commentJson.sentiment;
+    const repliesManager = RepliesManager.toRepliesManager(commentJson.repliesManager);
+    return new Comment(commenterName, commentMsg, commentSentiments, repliesManager);
```

Comment:
Sorry about the confusion. My intention is to have `repliesManager` manage the `Reply`s. I need its `id` because I want to set `Reply`'s ancestor to `RepliesManager`. Since this is confusing, how do you think I can better implement it? Thanks!

Reply:
If I'm understanding correctly, each comment has a corresponding RepliesManager, identified by id. I'm wondering that instead of keeping the id in replied manager, can we have an id for each comment instead? The comment json could simply contain a list of comments, and as of now I don't see the importance of RepliesManager, as it's only handling the logic of adding comment, which I feel that could be inside `Comment` class.

For example, Comment class will have the following
```
public class Comment {
  private String id;
  private String commenterName;
  private String commentMsg;
  private float sentiment;
  private List<String> replies;

  public Comment(String id, String commenterName, String commentMsg, float sentiment, List<String> replies) {
    this.id = id;
    this.commenterName = commenterName;
    this.commentMsg = commentMsg;
    this.sentiment = sentiment;
    this.replies = replies;
  }

  // Constructs comment fro entity
  public Comment(Entity commentEntity) {
     ...
  }

  public void addReply(...) {
    ...
  }
 }
```

Reply:
And then on the front-end side, we can keep a similar structure for comment, which would simplify the current structure a log.

Reply:
Thanks for the suggestion! (: This was what I had in mind. However, after brooding over it, I realised that I don't necessary want `Comment` to know about the replies. I don't think that it needs to know about it. Instead, for this application, `Comment` simply needs to know the number of positive/negative replies. I would like to use the sentiments from these replies and the comment's sentiment as data for this app. The number of replies with positive/negative sentiment will be managed by `RepliesManager`.  

What do you think? 😅

Reply:
Hmm I don't quite see what you mean.

The current logic is that Comment holds a RepliesManager, which is capable of addReply. If someone wants to addReply to the RepliesManager, we would do `comment.getRepliesManger().addReply()`. From the outside point of view, the knowledge of replies is still scoped to comment, and so far I don't quite see the need for ReplyManager, besides the fact that it doesn't actually have any logic for *managing*. It's only holds some replies data. wdyt?

Reply:
Sounds good! But this means that `Comment` will own the `ReplyForm` as well, won't it? Presently, `ReplyForm` is coupled to `RepliesManager`.

Reply:
Hmmm I was talking about the server side setup here... See my example code above, it's in java :) But I should have commented under the java code instead.

Reply:
Oh. HAHA. Sorry. I didn't think too much about the language. I just took it as pseudocode. 😅